sample/src/main/java/com/airbnb/airmapview/sample/MainActivity.java

```diff
@@ -1,218 +1,114 @@
-        airMapInterface = new WebAirMapViewBuilder().withOptions(new GoogleChinaMapType()).build();
-        break;
-      case R.id.action_clear_logs:
-        textLogs.setText("");
-        break;
-      default:
-        break;
-    }
+    Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+    DrawerLayout drawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+    ActionBarDrawerToggle drawerToggle = new ActionBarDrawerToggle(this, drawerLayout, toolbar,
+        R.string.drawer_open, R.string.drawer_close);
+    drawerLayout.setDrawerListener(drawerToggle);
+    drawerToggle.syncState();
 
-    if (airMapInterface != null) {
-      map.initialize(getSupportFragmentManager(), airMapInterface);
-    }
+    mListView = (ListView) findViewById(android.R.id.list);
+    mListView.setAdapter(mDemoAdapter);
+    mListView.setOnItemClickListener(this);
 
-    return super.onOptionsItemSelected(item);
-  }
+    addDemo("Default", DefaultDemoFragment.class);
+    addDemo("Ground Overlay", GroundOverlayDemoFragment.class);
 
-  @Override public void onCameraChanged(LatLng latLng, int zoom) {
-    appendLog("Map onCameraChanged triggered with lat: " + latLng.latitude + ", lng: "
-        + latLng.longitude);
+    startDemo(mDemos.get(0));
   }
 
-  @Override public void onMapInitialized() {
-    appendLog("Map onMapInitialized triggered");
-    final LatLng airbnbLatLng = new LatLng(37.771883, -122.405224);
-    addMarker("Airbnb HQ", airbnbLatLng, 1);
-    addMarker("Performance Bikes", new LatLng(37.773975, -122.40205), 2);
-    addMarker("REI", new LatLng(37.772127, -122.404411), 3);
-    addMarker("Mapbox", new LatLng(37.77572, -122.41354), 4);
-    map.animateCenterZoom(airbnbLatLng, 10);
-
-    // Add Polylines
-    LatLng[] latLngs = {
-        new LatLng(37.77977, -122.38937),
-        new LatLng(37.77811, -122.39160),
-        new LatLng(37.77787, -122.38864) };
-
-    map.addPolyline(new AirMapPolyline(Arrays.asList(latLngs), 5));
-
-    // Add Polygons
-    LatLng[] polygonLatLngs = {
-            new LatLng(37.784, -122.405),
-            new LatLng(37.784, -122.406),
-            new LatLng(37.785, -122.406),
-            new LatLng(37.785, -122.405)
-    };
-    map.addPolygon(new AirMapPolygon.Builder().add(polygonLatLngs).strokeWidth(3.f).build());
-
-    // Add Circle
-    map.drawCircle(new LatLng(37.78443, -122.40805), 1000);
-
-    // enable my location
-    map.setMyLocationEnabled(true);
-  }
-
-  private void addMarker(String title, LatLng latLng, int id) {
-    map.addMarker(new AirMapMarker.Builder()
-        .id(id)
-        .position(latLng)
-        .title(title)
-        .iconId(R.mipmap.icon_location_pin)
-        .build());
+  private void startDemo(Demo demo) {
+    try {
+      getSupportFragmentManager().beginTransaction().replace(R.id.content, demo.fragment.newInstance()).commit();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
can you remove this try/catch? it could cause the sample app to get in a weird state